skrub/_table_vectorizer.py

```diff
@@ -32,79 +29,41 @@
+    X : pandas.DataFrame
     """
-    if len(date_column) == 0:
-        return
-    date_column_sample = date_column.dropna().sample(
-        frac=min(n_trials / len(date_column), 1), random_state=42
-    )
-    # try to infer the date format
-    # see if either dayfirst or monthfirst works for all the rows
-    with warnings.catch_warnings():
-        # pandas warns when dayfirst is not strictly applied
-        warnings.simplefilter("ignore")
-        date_format_monthfirst = date_column_sample.apply(
-            lambda x: guess_datetime_format(x)
-        )
-        date_format_dayfirst = date_column_sample.apply(
-            lambda x: guess_datetime_format(x, dayfirst=True),
-        )
-    # if one row could not be parsed, return None
-    if date_format_monthfirst.isnull().any() or date_format_dayfirst.isnull().any():
-        return
-    # even with dayfirst=True, monthfirst format can be inferred
-    # so we need to check if the format is the same for all the rows
-    elif date_format_monthfirst.nunique() == 1:
-        # one monthfirst format works for all the rows
-        # check if another format works for all the rows
-        # if so, raise a warning
-        if date_format_dayfirst.nunique() == 1:
-            # check if monthfirst and dayfirst haven't found the same format
-            if date_format_monthfirst.iloc[0] != date_format_dayfirst.iloc[0]:
-                warnings.warn(
-                    f"""
-                    Both {date_format_monthfirst.iloc[0]} and
-                    {date_format_dayfirst.iloc[0]} are valid formats for the dates in
-                    column '{date_column.name}'.
-                    Format {date_format_monthfirst.iloc[0]} will be used.
-                    """,
-                    UserWarning,
-                    stacklevel=2,
-                )
-        return date_format_monthfirst.iloc[0]
-    elif date_format_dayfirst.nunique() == 1:
-        # only this format works for all the rows
-        return date_format_dayfirst.iloc[0]
-    else:
-        # more than two different formats were found
-        # TODO: maybe we could deal with this case
-        return
+    columns = X.columns
+    X_out = X.copy()
+    for column in columns:
+        X_col = X[column]
+        if not is_datetime64_any_dtype(X_col):
+            # We don't use errors="ignore" because it casts string
+            # and categories to object dtype in Pandas < 2.0.
+            # TODO: replace 'raise' by 'ignore' and remove the exception
+            # catching when the minimum pandas version of skrub is 2.0.
+            try:
+                X_out[column] = pd.to_numeric(X_col, errors="raise")
+            except (ValueError, TypeError):
+                pass
+    return X_out
 
 
-def _has_missing_values(df: pd.DataFrame | pd.Series) -> bool:
+def _has_missing_values(df):
     """
     Returns True if `array` contains missing values, False otherwise.
```

Comment:
Nice catch. This is indeed for a series. Do you prefer:
a. reverting to `any` to keep it generalistic
b. removing the function since it's a one-liner, and we won't need to document it

Reply:
I think b because it is only used in 2 places, and adds an indirection without making the code shorter, encapsulating something that might change or anything complex